jenkins/pipeline.groovy

```diff
@@ -115,122 +115,128 @@ pipeline {
-                        until aws s3 cp --no-progress --acl public-read sources/results/*.tar.gz s3://ps-build-cache/${BUILD_TAG}/binary.tar.gz; do
-                            sleep 5
-                        done
-                    else
-                        echo cannot find compiled archive
-                        exit 1
-                    fi
-                '''
+                        if [[ -f \$(ls sources/results/*.tar.gz | head -1) ]]; then
+                            until aws s3 cp --no-progress --acl public-read sources/results/*.tar.gz s3://ps-build-cache/${BUILD_TAG}/binary.tar.gz; do
+                                sleep 5
+                            done
+                        else
+                            echo cannot find compiled archive
+                            exit 1
+                        fi
+                    '''
+                    }
                 }
             }
         }
         stage('Archive Build') {
             agent { label 'micro-amazon' }
             steps {
-                retry(3) {
-                deleteDir()
-                sh '''
-                    aws s3 cp --no-progress s3://ps-build-cache/${BUILD_TAG}/build.log.gz ./build.log.gz
-                    gunzip build.log.gz
-                '''
-                warnings canComputeNew: false, canResolveRelativePaths: false, categoriesPattern: '', defaultEncoding: '', excludePattern: '', healthy: '', includePattern: '', messagesPattern: '', parserConfigurations: [[parserName: 'GNU C Compiler 4 (gcc)', pattern: 'build.log']], unHealthy: ''
+                timeout(time: 60, unit: 'MINUTES')  {
+                    retry(3) {
+                        deleteDir()
+                        sh '''
+                            aws s3 cp --no-progress s3://ps-build-cache/${BUILD_TAG}/build.log.gz ./build.log.gz
+                            gunzip build.log.gz
+                        '''
+                        warnings canComputeNew: false, canResolveRelativePaths: false, categoriesPattern: '', defaultEncoding: '', excludePattern: '', healthy: '', includePattern: '', messagesPattern: '', parserConfigurations: [[parserName: 'GNU C Compiler 4 (gcc)', pattern: 'build.log']], unHealthy: ''
+                    }
                 }
             }
         }
         stage('Test') {
             agent { label LABEL }
             steps {
-                retry(3) {
-                git branch: '8.0', url: 'https://github.com/Percona-Lab/ps-build'
-                    withCredentials([string(credentialsId: 'MTR_VAULT_TOKEN', variable: 'MTR_VAULT_TOKEN')]) {
-                        sh '''
-                            sudo git reset --hard
-                            sudo git clean -xdf
-                            rm -rf sources/results
-                            sudo git -C sources reset --hard || :
-                            sudo git -C sources clean -xdf   || :
+                timeout(time: pipeline_timeout, unit: 'HOURS')  {
```

Comment:
with `-DWITH_VALGRIND` pipeline_timeout = 144 but it has no sense because the general timeout is 3 days

Reply:
thank you, fixed